docs/user-manual/design/fpp-json-dict.md

```diff
@@ -729,39 +789,130 @@ product record Record1: U32 id 0x102
 
 Example FPP model with JSON representation:
 ```
-@ Container 0
-@ Implied id is 0x100
-product container Container0
+module M {
+
+  active component Component1 { 
+
+    @ Container 0
+    product container Container0 id 0x07
+
+    @ Container 1
+    product container Container1 id 0x08
+
+    @ Container 2
+    product container Container2 id 0x09 default priority 10
+
+  }
 
-@ Container 1
-product container Container1 id 0x102
+  instance C1: Component1 base id 0x100 \
+    queue size 10
+
+  topology T {
+    instance C1
+  }
+
+}
 
-@ Container 2
-@ Implied id is 0x103
-product container Container2 default priority 10
 ```
 
 ```json
 [
     {
-       "name": "M1.Container0",
-       "annotation": "Container 0\nImplied id is 0x100",
-       "id": 256,
+       "name": "M.C1.Container0",
+       "annotation": "Container 0",
+       "id": 263,
     },
     {
-        "name": "M1.Container1",
+        "name": "M.C1.Container1",
         "annotation": "Container 1",
-        "id": 258,
+        "id": 264,
     },
     {
-        "name": "M1.Container2",
-        "annotation": "Container 2\nImplied id is 0x103",
-        "id": 3,
+        "name": "M.C1.Container2",
+        "annotation": "Container 2",
+        "id": 265,
         "defaultPriority": 259
     }
 ]
 ```
 
+## Telemetry Packet Sets
+
+### Telemetry Packets
+| Field | Description | Options | Required |
+| ----- | ----------- | ------- | -------- |
+| `name` | Name of the telemetry packet | **String** | true |
+| `id` | Numeric identifier of the packet | **Number** | true |
+| `group` | Packet group number | **Number** | true |
+| `members` | Telemetry Channels in the packet | Array of Fully Qualified Names of [Telemetry Channels](#telemetry-channels) | true
+
+### Telemetry Packet Sets
+| Field | Description | Options | Required |
+| ----- | ----------- | ------- | -------- |
+| `name` | Name of the telemetry packet set | **String** | true |
+| `members` | Telemetry Packets in the set | Array of [Telemetry Packets](#telemetry-packets) | true |
+| `omitted` | Telemetry Channels omitted from the set | Array of Fully Qualified Names of [Telemetry Channels](#telemetry-channels) | true |
+
+
+Example FPP model with JSON representation:
+```
+module M {
+
+  active component C {
+    @ Telemetry channel 0
+    telemetry Channel0: U32 id 0x00
+
+    @ Telemetry channel 1
+    telemetry Channel1: U32 \
+      id 0x01 \
+      update on change
+
+    @ Telemetry channel 2
+    telemetry Channel2: F64 \
+      id 0x02 \
+      format "{.3f}"
+  }
+
+  instance C1: C base id 0x100 \
+    queue size 10
+
+  topology T {
+    instance C1
+
+    telemetry packets Packets {
+      packet P1 id 0 level 0 {
+        M.C1.Channel0
+        M.C1.Channel1
+      }
+    } omit {
+      M.C1.Channel2
+    }
+  }
+}
+```
+
+```json
+[
+  {
+      "name" : "Packets",
+      "members" : [
+        {
+          "name" : "P1",
```

Comment:
I would expect the packet name to be qualified, shouldn't I?  i.e. `M.T.P1`.

Reply:
Agreed that it would likely make things easier on the GDS to have a qualified name

Reply:
We can make the name qualified if you want, but to me that seems unusual. The packet name is part of the packet set definition, similarly to the way a struct member name is part of a struct definition. In the past the convention we've used is that definitions use simple names and uses use qualified names. The issue with uses is that without the explicit qualification you have to do some name lookup. With definitions there is no such issue -- the name is qualified by the definition in which it appears.

Reply:
> In the past the convention we've used is that definitions use simple names and uses use qualified names.

Unless the definition is stripped out of the qualifying context. So for example (I think) we use qualified names for the names of struct types but simple names for the members. We have to use qualified names for the types because the enclosing module context is stripped away when passing from the FPP source to the dictionary.

Here I think we need the module (and topology) context for the packet set. We could also add that to every packet in the set, but to me it seems overly verbose and maybe inconsistent.

Reply:
So in the example I think

* The packet set name should be `M.T.Packets`
* The packet name could be `M.T.Packets.P1` but I would argue against this -- I would argue that `P1` is more like a struct member.

Reply:
However, I think there's also an argument that we don't need the qualifier for the packet set (so the example is correct as is) because everything in the dictionary pertains to that topology. So the topology qualifier is redundant (indeed, everything in the dictionary could be qualified that way).

Reply:
When we generate C++ code, the packet sets will become classes, and their names will be appropriately qualified, because in C++ we don't assume that the code exists in an outer topology context.

Reply:
In my current implementation, telemetry packet set names are not fully qualified. This is also the case for the 
 set members/telemetry packet names. @LeStarch @thomas-bc is there a specific use case you have in mind for the GDS that would benefit from making these names fully qualified? After talking to Rob, we were thinking of adding a topology name field to the dictionary metadata (for ex: `topologyName`: `M.T`) which would make that information available in the dictionary.

Reply:
> we were thinking of adding a topology name field to the dictionary metadata

The latest thinking is that the `deployment name` field should hold the fully-qualified topology name.

Reply:
> When we generate C++ code, the packet sets will become classes

Note: In the current C++ implementation, the packet sets are stored as instances of a generic class called `Svc::TlmPacketizerPacketList`. The names are used for generating local variables that are hidden inside the implementation. There is no name qualification, and the names are not globally visible at all. This is true for both packets and packet sets.

Reply:
Works for me.